modules/src/ics02_client/trust_threshold.rs

```diff
@@ -0,0 +1,130 @@
+//! IBC Domain type definition for [`TrustThreshold`]
+//! represented as a fraction with valid values in the
+//! range `[0, 1)`.
+
+use std::{convert::TryFrom, fmt};
+
+use serde::{Deserialize, Serialize};
+use tendermint_proto::Protobuf;
+
+use ibc_proto::ibc::lightclients::tendermint::v1::Fraction;
+use tendermint::trust_threshold::TrustThresholdFraction;
+
+use crate::ics02_client::error::Error;
+
+/// [`TrustThreshold`] defines the level of trust that a client has
+/// towards a set of validators of a chain.
+///
+/// A trust threshold is represented as a fraction, i.e., a numerator and
+/// and a denominator.
+/// A typical trust threshold is 1/3 in practice.
+/// This type accepts even a value of 0, (numerator = 0, denominator = 0),
+/// which is used in the client state of an upgrading client.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TrustThreshold {
+    numerator: u64,
+    denominator: u64,
+}
+
+impl TrustThreshold {
+    /// Constant for a trust threshold of 1/3.
+    pub const ONE_THIRD: Self = Self {
+        numerator: 1,
+        denominator: 3,
+    };
+
+    /// Constant for a trust threshold of 2/3.
+    pub const TWO_THIRDS: Self = Self {
+        numerator: 2,
+        denominator: 3,
+    };
+
+    /// Constant for a trust threshold of 0/0.
+    pub const ZERO: Self = Self {
+        numerator: 0,
+        denominator: 0,
+    };
+
+    /// Instantiate a TrustThreshold with the given denominator and
+    /// numerator.
+    ///
+    /// The constructor succeeds if long as the resulting fraction
+    /// is in the range`[0, 1)`.
+    pub fn new(numerator: u64, denominator: u64) -> Result<Self, Error> {
+        // The two parameters cannot yield a fraction that is bigger or equal to 1
+        if (numerator > denominator)
+            || (denominator == 0 && numerator != 0)
+            || (numerator == denominator && numerator != 0)
```

Comment:
I don't think the parens are necessary but we can keep them if you think it improve readability.

Reply:
Hmm, the parens did help me while writing this code.